src/components/UserProfile/BlueSquareLayout.jsx

```diff
@@ -77,6 +85,42 @@ const BlueSquareLayout = props => {
     setShow(false);
   }, []);
 
+  // This handler is used for Info Modal, that open when button "Can't Schedule Time Off" is clicked 
+  const openInfoModal = useCallback(() => { 
+    setShowInfoModal(true);
+  }, []);
+// This handler is used to close Info Modal, 
+  const closeInfoModal = useCallback(() => {
+    setShowInfoModal(false);
+  }, []);
+  // This handler is used for Explanation Modal, that open when <a>Click to learn why </a> is clicked 
+  const openExplanationModal = useCallback(() => { 
+    setShowExplanation(true);
+  }, []);
+// This handler is used to close Info Modal, 
+  const closeExplanationModal  = useCallback(() => {
+    setShowExplanation(false);
+  }, []);
+
+
+//  This useEffect will check for any changes in the number of infringements
+  useEffect(()=>{
```

Comment:
This is my first PR review, so I'm getting up to speed understanding things, so feel free to push back. But is this useEffect necessary? Can you not derive the number of infringements by the length of the infringement array? Additionally, this useEffect is likely throwing a linter error due to the use of the useProfile object. I get you left it empty so it only runs once when the component renders, but the dependancy array should be populated with the userProfile, and if this causes problems it's probably better to not use the effect.

Reply:
Hi there, @ilyaflaks! Thank you so much for pointing that out. I just wanted to clarify for you, @F-Adam-B, that the reason I added the useEffect hook was because I needed to check for infringements after the component had rendered. The existing code already takes care of checking the user's role for permissions. I really appreciate you taking the time to review my code, and I'm always open to any suggestions you may have. I'm still learning, so any feedback is super valuable to me! Thanks again!